File: app/src/main/java/com/wb/logistics/utils/prefs/SharedWorker.java
package com.wb.logistics.utils.prefs;

import androidx.annotation.NonNull;

public interface SharedWorker {

    boolean load(@NonNull String key, boolean defValue);

    void save(@NonNull String key, boolean value);

    @NonNull
    String load(@NonNull String key, @NonNull String defValue);

    void save(@NonNull String key, @NonNull String value);

    void delete(@NonNull String... keys);

}

File: app/src/main/java/com/wb/logistics/utils/prefs/SharedWorker.kt
package com.wb.logistics.utils.prefs

interface SharedWorker {
    fun load(key: String, defValue: Boolean): Boolean
    fun save(key: String, value: Boolean)
    fun load(key: String, defValue: String): String
    fun save(key: String, value: String)
    fun <T : Any> save(key: String, value: T)
    fun <T> load(key: String, serializeClass: Class<T>): T
    fun isAllExists(vararg keys: String): Boolean
    fun delete(vararg keys: String)
}